server/routerlicious/packages/protocol-base/src/protocol.ts

```diff
@@ -116,22 +149,49 @@ export class ProtocolOpHandler {
         this.minimumSequenceNumber = message.minimumSequenceNumber;
         this.sequenceNumber = message.sequenceNumber;
 
+        this.cachedProtocolState.minimumSequenceNumber = this.minimumSequenceNumber;
+        this.cachedProtocolState.sequenceNumber = this.sequenceNumber;
+
         // Notify the quorum of the MSN from the message. We rely on it to handle duplicate values but may
         // want to move that logic to this class.
-        immediateNoOp = this.quorum.updateMinimumSequenceNumber(message) || immediateNoOp;
+        const updateMsnResult = this.quorum.updateMinimumSequenceNumber(message);
+        if (updateMsnResult) {
+            if (updateMsnResult.immediateNoOp) {
+                immediateNoOp = true;
+            }
+
+            if (updateMsnResult.proposals) {
+                // proposals are changing
+                this.cachedProtocolState.proposals = undefined;
```

Comment:
I feel like this code is asking for caches to be in quorum (the part that is relevant to quorum). Otherwise we are plumbing back and forth knowledge if quorum state changed or not.